flink-streaming-java/src/test/java/org/apache/flink/streaming/environment/ExecutorDiscoveryAndJobClientTest.java

```diff
@@ -69,19 +88,39 @@ private JobExecutionResult executeTestJobBasedOnConfig(final Configuration confi
 	 */
 	public static class IDReportingExecutorFactory implements ExecutorFactory {
 
-		public static final String ID = "test-executor-A";
-
 		@Override
 		public boolean isCompatibleWith(@Nonnull Configuration configuration) {
-			return ID.equals(configuration.get(DeploymentOptions.TARGET));
+			return EXEC_NAME.equals(configuration.get(DeploymentOptions.TARGET));
 		}
 
 		@Override
 		public Executor getExecutor(@Nonnull Configuration configuration) {
 			return (pipeline, executionConfig) -> {
 				final Map<String, OptionalFailure<Object>> res = new HashMap<>();
-				res.put(DeploymentOptions.TARGET.key(), OptionalFailure.of(ID));
-				return new JobExecutionResult(new JobID(), 12L, res);
+				res.put(DeploymentOptions.TARGET.key(), OptionalFailure.of(EXEC_NAME));
+
+				return CompletableFuture.completedFuture(new JobClient(){
```

Comment:
Here is another instance which we can replace with a single subclass `TestingJobClient`.

Reply:
I notice that it is less than awesome we introduce `TestingJobClient` in `flink-test-utils` since it has a scala suffix. Revoke my comment for that for now. We can rethink a scala-free test-utils later.